fxa/plugins/requests.py

```diff
@@ -33,24 +36,44 @@ class FxABrowserIDAuth(AuthBase):
 
     """
     def __init__(self, email, password, audience=None, with_client_state=False,
-                 server_url=core.DEFAULT_SERVER_URL):
+                 server_url=core.DEFAULT_SERVER_URL, cache=True,
+                 ttl=None):
         self.email = email
         self.password = password
         self.audience = audience
         self.with_client_state = with_client_state
         self.server_url = server_url
 
-    def __call__(self, request):
-        client = core.Client(server_url=self.server_url)
-        session = client.login(self.email, self.password, keys=True)
+        self.cache = cache
+        if self.cache is True:
+            if ttl is None:
+                ttl = DEFAULT_CACHE_EXPIRY - 1
+            self.cache = oauth.MemoryCache(ttl)
```

Comment:
Perhaps worth pulling `MemoryCache` out into a shared utility rather than using it directly from the `oauth` submodule.